mw/Map.d.ts

```diff
@@ -24,10 +24,7 @@ declare global {
              * @see https://doc.wikimedia.org/mediawiki-core/master/js/#!/api/mw.Map-method-get
              */
             get(): V;
-            get<S extends Array<keyof V>>(
-                selection: S,
-                fallback?: any
-            ): Pick<V, S extends Array<infer SS> ? SS : never>;
+            get<S extends keyof V>(selection: S[], fallback?: any): Pick<V, S>;
```

Comment:
When using `get` with and array of const values, the compiler was not able to infer the result type correctly (e.g. `{"k": string}` instead of `{"k": "v1" | "v2"}`), so the type variable is siplified for it to be infered correctly.

Reply:
How about this? https://github.com/AnYiEE/types-mediawiki-renovate/commit/6df693ab18bb3008051dbbd1b49ca219683fdd0b

Reply:
> How about this? [AnYiEE@6df693a](https://github.com/AnYiEE/types-mediawiki-renovate/commit/6df693ab18bb3008051dbbd1b49ca219683fdd0b)

I'm sorry, I'm not sure I understand what these additional type variables are used for, and I can't find much information about the error message mentioned in the commit description. Do you have an example of this error or of an usage of these type variables?

Reply:
> > How about this? [AnYiEE@6df693a](https://github.com/AnYiEE/types-mediawiki-renovate/commit/6df693ab18bb3008051dbbd1b49ca219683fdd0b)
> 
> I'm sorry, I'm not sure I understand what these additional type variables are used for, and I can't find much information about the error message mentioned in the commit description. Do you have an example of this error or of an usage of these type variables?

Passing the type to `mw.messages.exists<types>()` allows you to obtain type checking and autocompletion functionality for a specific object, without having to wait until runtime to know certain things. In previous versions, it was not possible to pass the type to `mw.messages.exists()` because `mw.messages` was already an instance of `mw.Map`.

```ts
type Msg = "a" | "b"

const msg: Record<Msg, string> = { ... }

mw.messages.exists<Msg>("a") // true
```